students/brian_warn/session05/except_exercise.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/python
+
+"""
+An exercise in playing with Exceptions.
+Make lots of try/except blocks for fun and profit.
+
+Make sure to catch specifically the error you find, rather than all errors.
+"""
+
+from except_test import fun, more_fun, last_fun
+
+
+# Figure out what the exception is, catch it and while still
+# in that catch block, try again with the second item in the list
+first_try = ['spam', 'cheese', 'mr death']
+
+try:
+    joke = fun(first_try[0])
+except NameError as ne:
+    print("fun() is not coded properly: " + str(ne))
+
+# Here is a try/except block. Add an else that prints not_joke
+try:
+    not_joke = fun(first_try[2])
+except SyntaxError as se:
+    print("Run Away!" + str(se))
+
+
+# What did that do? You can think of else in this context, as well as in
+# loops as meaning: "else if nothing went wrong"
+# (no breaks in  loops, no exceptions in try blocks)
+
+# Figure out what the exception is, catch it and in that same block
+#
+# try calling the more_fun function with the 2nd language in the list,
+# again assigning it to next_joke.
+#
+# If there are no exceptions, call the more_fun function with the last
+# language in the list regardless of whether there was an exception
+
+# Finally, while still in the try/except block and regardless of whether
```

Comment:
REad this again carefully -- last_fun() does not raise an Exception.

rather, we want you to call it after the other one, whether an Exception or not was raised.

There is a keyword in a a try: suite that means: "run this code at the end always" -- put the last_fun() call in that block.

yes, I'm being a bit mysterious on purpose :-)

Reply:
@PythonCHB, I think I see now what you have in mind here. I'll put in place the change when I'm back on Sunday night.

Reply:
Sounds good.

If you can think of clearer wording, let us know and we can update the exercise for the next time.